### PR TITLE
feat(plugin-chart-echarts): Echarts Waterfall

### PR DESCRIPTION
**Pull request by stephenLYZ**
🏆 Enhancements

Waterfall charts are useful for understanding how an initial value (like net income) is affected by a series of positive and negative changes. 

For example, we can see the trend of personal annual income. Here we use the year as the `series`.
image

Moreover, we can use the value in `breakdowns` to add additional data to the chart. It adds the contributors to increases or decreases for each series.
image

This pr refers to #705 (Thanks for Simcha's work.

### CODE REVIEW

**Comment by xiezhongfu** [comment]
Thanks stephen.
Core-code is LGTM for me. Next, I will review the code in detail.

**Review by zhaoyongjie — COMMENTED** [review]
Awesome works! Leave some issues.

1. the `null` value needs to be handled in tooltip
image


2. please use `seriese` control as category, and need to DND support
image

3. I can't hidden `metric value` and `tooltip` on the viz

image

4. the `Breakdown` is broken
image

File: plugins/plugin-chart-echarts/src/Waterfall/index.ts [hunk 1]
```diff
@@ -0,0 +1,55 @@
...
+   * It is worth noting that `buildQuery` and is optional, and only needed for
+   * advanced visualizations that require either post processing operations
+   * (pivoting, rolling aggregations, sorting etc) or submitting multiple queries.
+   */
+  constructor() {
+    super({
+      buildQuery,
+      controlPanel,
+      loadChart: () => import('./EchartsWaterfall'),
+      metadata: new ChartMetadata({
```
Comment (zhaoyongjie) [inline plugins/plugin-chart-echarts/src/Waterfall/index.ts:46]: needs category property here. Might `Ranking`  is  is an appropriate category

Comment (rusackas) [reply plugins/plugin-chart-echarts/src/Waterfall/index.ts:46]: `Evolution` seems reasonable to me.

**Comment by stephenLYZ** [comment]
> Awesome works! Leave some issues.
> 
> 1. the `null` value needs to be handled in tooltip
>    image
> 2. please use `seriese` control as category, and need to DND support
>    image
> 3. I can't hidden `metric value` and `tooltip` on the viz
> 
> image
> 
> 1. the `Breakdown` is broken
> 
> image

1. fixed.
2. fixed.
3. add `show_value` control and `rich_tooltip` control for this chart. I'm not sure 
4. The logic of the previous breakdown was to hide the breakdown of the first series and only show the total (a reference to powerBI?). Maybe it is strange because it will cause the data and the chart to not correspond, I removed this special logic first and need to discuss whether this special logic is needed.

**Comment by villebro** [comment]
The codebase on this repo has been moved to the main [Apache Superset repo](https://github.com/apache/superset), and consequently the repo is in the process of being archived. See the Superset Improvement Proposal for details: https://github.com/apache/superset/issues/13013 . While all currently open issues and PRs will be closed, we encourage you to reopen this PR on the main repo, which should be as simple as moving over any code changes as follows:
- core packages: `packages/**` (this repo) -> `superset-frontend/packages/**` (main repo)
- plugins: `plugins/**` (this repo) -> `superset-frontend/plugins/**` (main repo)

If you need help with the migration, please post a message on the SIP or reach out on the [community Slack](https://join.slack.com/t/apache-superset/shared_invite/zt-uxbh5g36-AISUtHbzOXcu0BIj7kgUaw).

**Pull request closed without merging**